### PR TITLE
Enable Ethos-U85 support in Vela

### PR DESCRIPTION
**Pull request by tom-arm**
Switch pinned version of Vela with Ethos-U85 support. Adapt the compiler arguments and add a compile_spec to support targetting Ethos-U85.

### CODE REVIEW

File: backends/arm/test/common.py [hunk 1]
```diff
@@ -137,7 +151,29 @@ def get_u55_compile_spec_unbuilt(
             "ethos-u55-128",
             system_config="Ethos_U55_High_End_Embedded",
             memory_mode="Shared_Sram",
-            extra_flags=None,
+            extra_flags="--debug-force-regor --output-format=raw",
```
Comment (digantdesai) [inline backends/arm/test/common.py:154]: do we always want to keep `debug-force-regor` on?

File: backends/arm/test/common.py [hunk 2]
```diff
@@ -137,7 +151,29 @@ def get_u55_compile_spec_unbuilt(
...
+    the compile spec before calling .build() to finalize it.
+    """
+    artifact_path = custom_path or tempfile.mkdtemp(prefix="arm_u85_")
+    compile_spec = (
+        ArmCompileSpecBuilder()
+        .ethosu_compile_spec(
+            "ethos-u85-128",
+            system_config="Ethos_U85_SYS_DRAM_Mid",
+            memory_mode="Shared_Sram",
+            extra_flags="--output-format=raw",
```
Comment (digantdesai) [inline backends/arm/test/common.py:176]: no debug flag here?

File: backends/arm/test/ops/test_add.py [hunk 3]
```diff
@@ -142,7 +149,9 @@ def test_add2_tosa_BI(self, operand1: torch.Tensor, operand2: torch.Tensor):
         test_data = (operand1, operand2)
         self._test_add_tosa_BI_pipeline(self.Add2(), test_data)
 
+    # TODO(MLETORCH-352) Remove @unittest.expectedFailure when bug fixed in Regor
```
Comment (digantdesai) [inline backends/arm/test/ops/test_add.py:152]: Can these tickets be mirrored on ET GH Issues? Or just description of the bug and corresponding MLE ticket in GH Issue?

File: examples/arm/setup.sh [hunk 4]
```diff
@@ -215,7 +215,7 @@ function setup_vela() {
     if [[ ! -e ethos-u-vela ]]; then
         git clone https://review.mlplatform.org/ml/ethos-u/ethos-u-vela
         repo_dir="${root_dir}/ethos-u-vela"
-        base_rev=7706c1281166e7611f4300ed26338087152a33c9
+        base_rev=d362f5443f67b1e6213a9d8f124edff758efac96
```
Comment (digantdesai) [inline examples/arm/setup.sh:218]: seems like we are regressing a bit on U55, is that a concern when we move to the new backend?

**Review by digantdesai — APPROVED** [review]
LGTM at a high level. Congratulations for getting Regor to this point. There has a been a lot of anticipation around this and us not able to see the progress made it suspenseful 😄.

Stamping it to indicate that the core is good. Let's please resolve comments before I merge.

File: backends/arm/test/common.py [hunk 5]
```diff
@@ -122,10 +125,21 @@ def get_u55_compile_spec(
     ).build()
 
 
+def get_u85_compile_spec(
```
Comment (digantdesai) [inline backends/arm/test/common.py:128]: Any reason we are not using this in any tests - at least till TOSA? Or it's in the future PRs?

File: backends/arm/test/common.py (hunk 1 above)
Comment (per) [reply backends/arm/test/common.py:154]: Yes, it's the way forward. We are enabling the new backend for U55 on the tests to sort out the last bits. The example still runs without the flag, so only affecting the unit tests at this point.

File: backends/arm/test/common.py (hunk 2 above)
Comment (per) [reply backends/arm/test/common.py:176]: Yes, that's right. For U85 it's the backend being used always.

File: examples/arm/setup.sh (hunk 4 above)
Comment (per) [reply examples/arm/setup.sh:218]: No, not really.

File: backends/arm/test/common.py (hunk 5 above)
Comment (per) [reply backends/arm/test/common.py:128]: Yes, and there is a separate PR being prepared to enable unit test for this compile spec as well.

File: examples/arm/setup.sh (hunk 4 above)
Comment (robell) [reply examples/arm/setup.sh:218]: How do you mean regressing Digant?
and Per, do you mean no we're not regressing or no we're not concerned? :)

Comment (digantdesai) [reply examples/arm/setup.sh:218]: Understood, thanks.

**Comment by digantdesai** [comment]
Rebase please?

File: backends/arm/test/ops/test_add.py (hunk 3 above)
Comment (tom-arm) [reply backends/arm/test/ops/test_add.py]: This shouldn't be needed anymore - as some of these TODOs have been removed. I have amended one of the TODOs in lieu of a GitHub issue

**Comment by tom-arm** [comment]
Sorry for the delayed response to the PR @digantdesai. Let me know if you need anything else

**Pull request closed without merging**